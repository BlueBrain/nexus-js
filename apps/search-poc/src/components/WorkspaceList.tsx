import * as React from 'react';
import './WorkspaceList.css';
import { Tabs } from 'antd';

const { TabPane } = Tabs;

type WorkspaceItem = {
  id: string;
  label: string;
  description: string;
};

const WorkspaceList: React.FunctionComponent<{
  items: WorkspaceItem[];
  onSelected: (id: string) => void;
  defaultActiveId?: string;
}> = ({ items, onSelected, defaultActiveId, children }) => {
  return (
    <div className="workspace-list">
      <Tabs
        defaultActiveKey={defaultActiveId}
        tabPosition={'left'}
        onChange={onSelected}
      >
        {items.map(({ label, description, id }) => (
          <TabPane
            tab={
              <div className="workspace-tab">
                <span className="title ellipsis">{label}</span>
                <p className="description fade">{description}</p>
              </div>
            }
            key={id}
          >
            {children}
          </TabPane>
        ))}
      </Tabs>
    </div>
  );
};

export default WorkspaceList;
