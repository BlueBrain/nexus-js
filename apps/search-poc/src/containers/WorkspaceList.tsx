import * as React from 'react';
import queryString from 'query-string';

import TabList from '../components/TabList';
import { HandleClickParams } from '../types';
import history from '../history';


function getWorkspaceConfig(
  id: string,
  configs: WorkspaceConfig[],
): WorkspaceConfig {
  return configs.find(config => config['@id'] === id) || configs[0];
}

type WorkspaceConfig = {
  '@id': string;
  label: string;
  description?: string;
};

const WorkspaceListContainer: React.FunctionComponent<{
  workspaceConfig: WorkspaceConfig[];
  onWorkspaceSelected: (workspaceId: string) => void;
  handleClick: (params: HandleClickParams) => void;
}> = ({ workspaceConfig, onWorkspaceSelected, handleClick }) => {

  // get active id from query string on mount (if any)
  const queryStringWorkspaceId =
    queryString.parse(history.location.search).workspace || '';

  // find out if we have a matching workspace with that id
  const activeId = getWorkspaceConfig(
    queryStringWorkspaceId.toString(),
    workspaceConfig,
  )['@id'];

  // format workspace data for Tablist component
  const configData = workspaceConfig.map(config => ({
    id: config['@id'],
    label: config.label,
    description: config.description || '',
  }));

  // trigger parent on mount with the active id
  React.useEffect(() => {
    onWorkspaceSelected(activeId);
    // eslint-disable-next-line
  }, []);

  return (
    <TabList
      position={'top'}
      items={configData}
      onSelected={workspaceId => {
        const activeWorkspace = workspaceId
          ? getWorkspaceConfig(workspaceId.toString(), workspaceConfig)
          : workspaceConfig[0];
        // we need to remove the active dashboard as we are changing workspaces
        const queryStrings = queryString.parse(history.location.search);
        // eslint-disable-next-line
        const { ['dashboard']: value, ...withoutDashboard } = queryStrings;
        const search = queryString.stringify({
          ...withoutDashboard,
          workspace: workspaceId,
        })

        handleClick({ search, type: 'dashboard' });
        onWorkspaceSelected(activeWorkspace['@id']);
      }}
      defaultActiveId={activeId}
    >
      {children}
    </TabList>
  );
};

export default WorkspaceListContainer;
