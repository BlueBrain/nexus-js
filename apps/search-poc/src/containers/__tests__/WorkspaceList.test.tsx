import * as renderer from "react-test-renderer";
import WorkspaceList from '../WorkspaceList';
import React from "react";
import { mount } from "enzyme";
import { BrowserRouter as Router } from 'react-router-dom';

describe('WorkSpaceList', () => {
    const workspaceConfigs = [{
        '@id': 'id',
        label: 'work space label',
        description: 'work space description'
    }] 

    it('WorkSpaceList renders correctly', () => {
        const props = { workspaceConfig :  workspaceConfigs, onWorkspaceSelected: jest.fn, history: jest.fn};
        const result = mount(<Router><WorkspaceList {...props} /></Router>);
        expect(result).toMatchSnapshot();
    });

});

