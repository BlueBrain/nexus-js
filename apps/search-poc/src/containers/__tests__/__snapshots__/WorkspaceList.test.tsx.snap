// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkSpaceList WorkSpaceList renders correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(WorkspaceListContainer)
      history={[Function]}
      onWorkspaceSelected={[Function]}
      workspaceConfig={
        Array [
          Object {
            "@id": "id",
            "description": "work space description",
            "label": "work space label",
          },
        ]
      }
    >
      <WorkspaceListContainer
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
        onWorkspaceSelected={[Function]}
        workspaceConfig={
          Array [
            Object {
              "@id": "id",
              "description": "work space description",
              "label": "work space label",
            },
          ]
        }
      >
        <TabList
          defaultActiveId="id"
          items={
            Array [
              Object {
                "description": "work space description",
                "id": "id",
                "label": "work space label",
              },
            ]
          }
          onSelected={[Function]}
          position="top"
        >
          <div
            className="tab-list"
          >
            <Tabs
              defaultActiveKey="id"
              hideAdd={false}
              onChange={[Function]}
              tabPosition="top"
            >
              <Tabs
                className="ant-tabs-line"
                defaultActiveKey="id"
                destroyInactiveTabPane={false}
                hideAdd={false}
                navWrapper={[Function]}
                onChange={[Function]}
                prefixCls="ant-tabs"
                renderTabBar={[Function]}
                renderTabContent={[Function]}
                style={Object {}}
                tabBarPosition="top"
                tabPosition="top"
              >
                <div
                  className="ant-tabs ant-tabs-top ant-tabs-line"
                  onScroll={[Function]}
                  style={Object {}}
                >
                  <TabBar
                    activeKey="id"
                    animated={true}
                    defaultActiveKey="id"
                    hideAdd={false}
                    key="tabBar"
                    navWrapper={[Function]}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    onTabClick={[Function]}
                    panels={
                      Array [
                        <TabPane
                          placeholder={null}
                          tab={
                            <div
                              className="tab-item"
                            >
                              <span
                                className="title ellipsis"
                              >
                                work space label
                              </span>
                              <p
                                className="description fade"
                              >
                                work space description
                              </p>
                            </div>
                          }
                        />,
                      ]
                    }
                    prefixCls="ant-tabs"
                    tabBarExtraContent={null}
                    tabBarPosition="top"
                    tabPosition="top"
                    type="line"
                  >
                    <ScrollableInkTabBar
                      activeKey="id"
                      animated={true}
                      className="ant-tabs-top-bar"
                      defaultActiveKey="id"
                      extraContent={null}
                      hideAdd={false}
                      inkBarAnimated={true}
                      navWrapper={[Function]}
                      nextIcon={
                        <span
                          className="ant-tabs-tab-next-icon"
                        >
                          <Icon
                            className="ant-tabs-tab-next-icon-target"
                            type="right"
                          />
                        </span>
                      }
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      onTabClick={[Function]}
                      panels={
                        Array [
                          <TabPane
                            placeholder={null}
                            tab={
                              <div
                                className="tab-item"
                              >
                                <span
                                  className="title ellipsis"
                                >
                                  work space label
                                </span>
                                <p
                                  className="description fade"
                                >
                                  work space description
                                </p>
                              </div>
                            }
                          />,
                        ]
                      }
                      prefixCls="ant-tabs"
                      prevIcon={
                        <span
                          className="ant-tabs-tab-prev-icon"
                        >
                          <Icon
                            className="ant-tabs-tab-prev-icon-target"
                            type="left"
                          />
                        </span>
                      }
                      tabBarExtraContent={null}
                      tabBarPosition="top"
                      tabPosition="top"
                      type="line"
                    >
                      <SaveRef>
                        <TabBarRootNode
                          activeKey="id"
                          animated={true}
                          className="ant-tabs-top-bar"
                          defaultActiveKey="id"
                          extraContent={null}
                          hideAdd={false}
                          inkBarAnimated={true}
                          navWrapper={[Function]}
                          nextIcon={
                            <span
                              className="ant-tabs-tab-next-icon"
                            >
                              <Icon
                                className="ant-tabs-tab-next-icon-target"
                                type="right"
                              />
                            </span>
                          }
                          onChange={[Function]}
                          onKeyDown={[Function]}
                          onTabClick={[Function]}
                          panels={
                            Array [
                              <TabPane
                                placeholder={null}
                                tab={
                                  <div
                                    className="tab-item"
                                  >
                                    <span
                                      className="title ellipsis"
                                    >
                                      work space label
                                    </span>
                                    <p
                                      className="description fade"
                                    >
                                      work space description
                                    </p>
                                  </div>
                                }
                              />,
                            ]
                          }
                          prefixCls="ant-tabs"
                          prevIcon={
                            <span
                              className="ant-tabs-tab-prev-icon"
                            >
                              <Icon
                                className="ant-tabs-tab-prev-icon-target"
                                type="left"
                              />
                            </span>
                          }
                          saveRef={[Function]}
                          style={Object {}}
                          tabBarExtraContent={null}
                          tabBarPosition="top"
                          tabPosition="top"
                          type="line"
                        >
                          <div
                            className="ant-tabs-bar ant-tabs-top-bar"
                            onKeyDown={[Function]}
                            role="tablist"
                            style={Object {}}
                            tabIndex="0"
                          >
                            <ScrollableTabBarNode
                              activeKey="id"
                              animated={true}
                              className="ant-tabs-top-bar"
                              defaultActiveKey="id"
                              extraContent={null}
                              getRef={[Function]}
                              hideAdd={false}
                              inkBarAnimated={true}
                              navWrapper={[Function]}
                              nextIcon={
                                <span
                                  className="ant-tabs-tab-next-icon"
                                >
                                  <Icon
                                    className="ant-tabs-tab-next-icon-target"
                                    type="right"
                                  />
                                </span>
                              }
                              onChange={[Function]}
                              onKeyDown={[Function]}
                              onNextClick={[Function]}
                              onPrevClick={[Function]}
                              onTabClick={[Function]}
                              panels={
                                Array [
                                  <TabPane
                                    placeholder={null}
                                    tab={
                                      <div
                                        className="tab-item"
                                      >
                                        <span
                                          className="title ellipsis"
                                        >
                                          work space label
                                        </span>
                                        <p
                                          className="description fade"
                                        >
                                          work space description
                                        </p>
                                      </div>
                                    }
                                  />,
                                ]
                              }
                              prefixCls="ant-tabs"
                              prevIcon={
                                <span
                                  className="ant-tabs-tab-prev-icon"
                                >
                                  <Icon
                                    className="ant-tabs-tab-prev-icon-target"
                                    type="left"
                                  />
                                </span>
                              }
                              saveRef={[Function]}
                              scrollAnimated={true}
                              tabBarExtraContent={null}
                              tabBarPosition="top"
                              tabPosition="top"
                              type="line"
                            >
                              <div
                                className="ant-tabs-nav-container"
                                key="container"
                              >
                                <span
                                  className="ant-tabs-tab-prev ant-tabs-tab-btn-disabled"
                                  onClick={null}
                                  onTransitionEnd={[Function]}
                                  unselectable="unselectable"
                                >
                                  <span
                                    className="ant-tabs-tab-prev-icon"
                                  >
                                    <Icon
                                      className="ant-tabs-tab-prev-icon-target"
                                      type="left"
                                    >
                                      <LocaleReceiver
                                        componentName="Icon"
                                      >
                                        <i
                                          aria-label="icon: left"
                                          className="anticon anticon-left ant-tabs-tab-prev-icon-target"
                                        >
                                          <IconReact
                                            className=""
                                            type="left-o"
                                          >
                                            <svg
                                              aria-hidden="true"
                                              className=""
                                              data-icon="left"
                                              fill="currentColor"
                                              focusable="false"
                                              height="1em"
                                              key="svg-left"
                                              viewBox="64 64 896 896"
                                              width="1em"
                                            >
                                              <path
                                                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                                                key="svg-left-svg-0"
                                              />
                                            </svg>
                                          </IconReact>
                                        </i>
                                      </LocaleReceiver>
                                    </Icon>
                                  </span>
                                </span>
                                <span
                                  className="ant-tabs-tab-next ant-tabs-tab-btn-disabled"
                                  onClick={null}
                                  unselectable="unselectable"
                                >
                                  <span
                                    className="ant-tabs-tab-next-icon"
                                  >
                                    <Icon
                                      className="ant-tabs-tab-next-icon-target"
                                      type="right"
                                    >
                                      <LocaleReceiver
                                        componentName="Icon"
                                      >
                                        <i
                                          aria-label="icon: right"
                                          className="anticon anticon-right ant-tabs-tab-next-icon-target"
                                        >
                                          <IconReact
                                            className=""
                                            type="right-o"
                                          >
                                            <svg
                                              aria-hidden="true"
                                              className=""
                                              data-icon="right"
                                              fill="currentColor"
                                              focusable="false"
                                              height="1em"
                                              key="svg-right"
                                              viewBox="64 64 896 896"
                                              width="1em"
                                            >
                                              <path
                                                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                                                key="svg-right-svg-0"
                                              />
                                            </svg>
                                          </IconReact>
                                        </i>
                                      </LocaleReceiver>
                                    </Icon>
                                  </span>
                                </span>
                                <div
                                  className="ant-tabs-nav-wrap"
                                >
                                  <div
                                    className="ant-tabs-nav-scroll"
                                  >
                                    <div
                                      className="ant-tabs-nav ant-tabs-nav-animated"
                                    >
                                      <TabBarTabsNode
                                        activeKey="id"
                                        animated={true}
                                        className="ant-tabs-top-bar"
                                        defaultActiveKey="id"
                                        extraContent={null}
                                        hideAdd={false}
                                        inkBarAnimated={true}
                                        navWrapper={[Function]}
                                        nextIcon={
                                          <span
                                            className="ant-tabs-tab-next-icon"
                                          >
                                            <Icon
                                              className="ant-tabs-tab-next-icon-target"
                                              type="right"
                                            />
                                          </span>
                                        }
                                        onChange={[Function]}
                                        onKeyDown={[Function]}
                                        onTabClick={[Function]}
                                        panels={
                                          Array [
                                            <TabPane
                                              placeholder={null}
                                              tab={
                                                <div
                                                  className="tab-item"
                                                >
                                                  <span
                                                    className="title ellipsis"
                                                  >
                                                    work space label
                                                  </span>
                                                  <p
                                                    className="description fade"
                                                  >
                                                    work space description
                                                  </p>
                                                </div>
                                              }
                                            />,
                                          ]
                                        }
                                        prefixCls="ant-tabs"
                                        prevIcon={
                                          <span
                                            className="ant-tabs-tab-prev-icon"
                                          >
                                            <Icon
                                              className="ant-tabs-tab-prev-icon-target"
                                              type="left"
                                            />
                                          </span>
                                        }
                                        renderTabBarNode={null}
                                        saveRef={[Function]}
                                        tabBarExtraContent={null}
                                        tabBarGutter={null}
                                        tabBarPosition="top"
                                        tabPosition="top"
                                        type="line"
                                      >
                                        <div>
                                          <div
                                            aria-disabled="false"
                                            aria-selected="true"
                                            className="ant-tabs-tab-active ant-tabs-tab"
                                            key="id"
                                            onClick={[Function]}
                                            role="tab"
                                            style={
                                              Object {
                                                "marginRight": null,
                                              }
                                            }
                                          >
                                            <div
                                              className="tab-item"
                                            >
                                              <span
                                                className="title ellipsis"
                                              >
                                                work space label
                                              </span>
                                              <p
                                                className="description fade"
                                              >
                                                work space description
                                              </p>
                                            </div>
                                          </div>
                                        </div>
                                      </TabBarTabsNode>
                                      <InkTabBarNode
                                        activeKey="id"
                                        animated={true}
                                        className="ant-tabs-top-bar"
                                        defaultActiveKey="id"
                                        extraContent={null}
                                        getRef={[Function]}
                                        hideAdd={false}
                                        inkBarAnimated={true}
                                        navWrapper={[Function]}
                                        nextIcon={
                                          <span
                                            className="ant-tabs-tab-next-icon"
                                          >
                                            <Icon
                                              className="ant-tabs-tab-next-icon-target"
                                              type="right"
                                            />
                                          </span>
                                        }
                                        onChange={[Function]}
                                        onKeyDown={[Function]}
                                        onTabClick={[Function]}
                                        panels={
                                          Array [
                                            <TabPane
                                              placeholder={null}
                                              tab={
                                                <div
                                                  className="tab-item"
                                                >
                                                  <span
                                                    className="title ellipsis"
                                                  >
                                                    work space label
                                                  </span>
                                                  <p
                                                    className="description fade"
                                                  >
                                                    work space description
                                                  </p>
                                                </div>
                                              }
                                            />,
                                          ]
                                        }
                                        prefixCls="ant-tabs"
                                        prevIcon={
                                          <span
                                            className="ant-tabs-tab-prev-icon"
                                          >
                                            <Icon
                                              className="ant-tabs-tab-prev-icon-target"
                                              type="left"
                                            />
                                          </span>
                                        }
                                        saveRef={[Function]}
                                        styles={Object {}}
                                        tabBarExtraContent={null}
                                        tabBarPosition="top"
                                        tabPosition="top"
                                        type="line"
                                      >
                                        <div
                                          className="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                                          key="inkBar"
                                        />
                                      </InkTabBarNode>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </ScrollableTabBarNode>
                          </div>
                        </TabBarRootNode>
                      </SaveRef>
                    </ScrollableInkTabBar>
                  </TabBar>
                  <Sentinel
                    key="sentinelStart"
                    setRef={[Function]}
                  >
                    <div
                      onKeyDown={[Function]}
                      role="presentation"
                      style={
                        Object {
                          "height": 0,
                          "overflow": "hidden",
                          "position": "absolute",
                          "width": 0,
                        }
                      }
                      tabIndex={0}
                    />
                  </Sentinel>
                  <TabContent
                    activeKey="id"
                    animated={true}
                    animatedWithMargin={true}
                    className="ant-tabs-top-content"
                    destroyInactiveTabPane={false}
                    key="tabContent"
                    onChange={[Function]}
                    prefixCls="ant-tabs"
                    tabBarPosition="top"
                  >
                    <div
                      className="ant-tabs-content ant-tabs-content-animated ant-tabs-top-content"
                      style={
                        Object {
                          "marginLeft": "0%",
                        }
                      }
                    >
                      <TabPane
                        active={true}
                        destroyInactiveTabPane={false}
                        key="id"
                        placeholder={null}
                        rootPrefixCls="ant-tabs"
                        tab={
                          <div
                            className="tab-item"
                          >
                            <span
                              className="title ellipsis"
                            >
                              work space label
                            </span>
                            <p
                              className="description fade"
                            >
                              work space description
                            </p>
                          </div>
                        }
                      >
                        <div
                          aria-hidden="false"
                          className="ant-tabs-tabpane ant-tabs-tabpane-active"
                          role="tabpanel"
                        >
                          <Sentinel
                            setRef={[Function]}
                          >
                            <div
                              onKeyDown={[Function]}
                              role="presentation"
                              style={
                                Object {
                                  "height": 0,
                                  "overflow": "hidden",
                                  "position": "absolute",
                                  "width": 0,
                                }
                              }
                              tabIndex={0}
                            />
                          </Sentinel>
                          <Sentinel
                            setRef={[Function]}
                          >
                            <div
                              onKeyDown={[Function]}
                              role="presentation"
                              style={
                                Object {
                                  "height": 0,
                                  "overflow": "hidden",
                                  "position": "absolute",
                                  "width": 0,
                                }
                              }
                              tabIndex={0}
                            />
                          </Sentinel>
                        </div>
                      </TabPane>
                    </div>
                  </TabContent>
                  <Sentinel
                    key="sentinelEnd"
                    setRef={[Function]}
                  >
                    <div
                      onKeyDown={[Function]}
                      role="presentation"
                      style={
                        Object {
                          "height": 0,
                          "overflow": "hidden",
                          "position": "absolute",
                          "width": 0,
                        }
                      }
                      tabIndex={0}
                    />
                  </Sentinel>
                </div>
              </Tabs>
            </Tabs>
          </div>
        </TabList>
      </WorkspaceListContainer>
    </withRouter(WorkspaceListContainer)>
  </Router>
</BrowserRouter>
`;
